app/controllers/api_keys_controller.rb

```diff
@@ -1,9 +1,8 @@
 class ApiKeysController < ApplicationController
   include ApiKeyable
-  before_action :redirect_to_signin, unless: :signed_in?
-  before_action :redirect_to_new_mfa, if: :mfa_required_not_yet_enabled?
-  before_action :redirect_to_settings_strong_mfa_required, if: :mfa_required_weak_level_enabled?
-  before_action :redirect_to_verify, unless: :password_session_active?
+
+  include SessionVerifiable
+  verify_session_before
```

Comment:
The class method obfuscates more than it simplifies, imho. A single before action method with the conditions as regular ruby code would be canonical, easier to read and easy find. I'd just drop it in ApplicationController so controllers that need a custom redirect could override `redirect_to_verify` and call super.

Reply:
the issue with collapsing to a single method is you lose the callback chain being halted as soon as one of them `redirect`s, meaning you have to put those conditionals into the method

Reply:
That doesn't seem like a big problem to me. Mostly I just removed commas and colons from the above code, then added the returns.

```ruby
   before_action :verify_session

   # ...

    def verify_session
      return redirect_to_signin unless signed_in?
      return redirect_to_new_mfa if mfa_required_not_yet_enabled?
      return redirect_to_settings_strong_mfa_required if mfa_required_weak_level_enabled?
      return redirect_to_verify unless password_session_active?
    end
```

The callback code builds a whole stack and is certainly slower than this one method, plus then we save a module and we don't need to go look up `verify_session_before` to see that it's actually just a set of before filters`.

Reply:
Oh, I looked closer at the filters themselves and they have their own conditions inside that might avoid the redirect. Unless you wanted to refactor that, then I agree with your original version for now.